src/lib/utils/redisCache.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-console */
```

Comment:
This file looks better, but I still have some questions. Async stuff can be quite confusing, so perhaps I just need to wake up further. See individual questions/comments below, and then some of my code brainstorming at the bottom.